timemachine/fe/free_energy.py

```diff
@@ -95,34 +95,40 @@ class PairBarResult:
     """Results of BAR analysis on L-1 adjacent pairs of states given a sequence of L states."""
 
     initial_states: List[InitialState]  # length L
-    bar_results: List[BarResult]  # length L - 1
+
+    # length L - 1 pairs of u_kln_by_component, BAR result
+    # NOTE: BAR result may be None if estimator fails to converge
+    bar_results: List[Tuple[NDArray, Optional[BarResult]]]
```

Comment:
If the BarResult object and the uklns are going to be separated, would prefer that there are two fields on PairBarResult rather than a `List[Tuple[NDArray, Optional[BarResult]]`